### ISSUE TITLE
Map art mode enhancements

### ISSUE THREAD

**Issue by someretical** [cabaletta/baritone#2591]
## Describe your suggestion
Currently building maps with baritone and frost walker is somewhat unreliable if you can't fit all materials into your inventory at once (which is almost always the case). By reliable, I mean that the process terminates way too often so you can't actually leave baritone AFK for long periods of time.

I would like to suggest a change to pathing in map art mode so that it takes frost walker into consideration. Basically, when searching for a path:
- Assume all water blocks that are in the y level beneath bottom layer of the schematic and NOT under a full block are frosted ice
- When actually travelling along the set path, wait for the water to turn into frosted ice if necessary
- Go to the nearest block that hasn't been filled regardless of whether there are other unfilled blocks near it (I have had issues with baritone stuck trying to path to unfilled remnants of maps when I have the right blocks in my inventory)
- When descending down to water level from a full block, do so via a non full block so that frosted ice can form
- Don't spend as much time trying to find the best path, if it gets there then it's probably good enough since the player might be AFK

When placing blocks:
- Commit to placing blocks in a cardinal direction, and when hitting a corner, go in another direction (doesn't matter if one has more blocks than another since all of them will have to be placed anyway). I'm suggesting this so that it is easier to find any unfilled spots. 
- If the player cannot place any blocks since the water is not frosted ice, re-path around using the closest non-full block in the schematic to stand on to generate frosted ice that the player can then walk across. If no such block is found, then terminate the process.

## Settings
The pathing taking frost walker boots into consideration could be an entirely new setting such as `assumeFrostWalker` to make planning a path much more efficient.

All the other changes could be implemented under the existing map art mode setting.

## Context
Basically suggesting all this so that the map art mode can be improved and it makes baritone more reliable.

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have not used any OwO's or UwU's in this issue.
